src/Uno.Extensions.Reactive.Messaging/MessengerExtensions.cs

```diff
@@ -0,0 +1,168 @@
+			CreateRecipient);
+
+		static Recipient<IState<TEntity>, TEntity, TKey> CreateRecipient(
+			IState<TEntity> state,
+			Func<TEntity, TKey> keySelector,
+			(IFeed<TOther> other, Func<TOther, TEntity, bool> predicate, IMessenger messenger) config)
+			=> new(state, config.messenger, keySelector, If<IState<TEntity>, TOther, TEntity, TKey>(config.other, config.predicate, StateExtensions.Update));
+	}
+
+	/// <summary>
+	/// Listen for <see cref="EntityMessage{TEntity}"/> on the given <paramref name="messenger"/>, matches it with another value and updates the <paramref name="listState"/> accordingly.
+	/// </summary>
+	/// <typeparam name="TOther">Type of the other value to validate.</typeparam>
+	/// <typeparam name="TEntity">Type of the value of teh state.</typeparam>
+	/// <typeparam name="TKey">Type of the identifier that uniquely identifies a <typeparamref name="TEntity"/>.</typeparam>
+	/// <param name="messenger">The messenger to listen for <see cref="EntityMessage{TEntity}"/></param>
+	/// <param name="listState">The list state to update.</param>
+	/// <param name="other">The other value to validate with the updated entity using <paramref name="predicate"/>.</param>
+	/// <param name="predicate">A predicted used with the updated entity to confirm that the change should by applied to the <paramref name="listState"/>.</param>
+	/// <param name="keySelector">A selector to get a unique identifier of a <typeparamref name="TEntity"/>.</param>
+	/// <returns>A disposable that can be used to unbind the state from the messenger.</returns>
+	public static IDisposable Observe<TOther, TEntity, TKey>(
+		this IMessenger messenger,
+		IListState<TEntity> listState,
+		IFeed<TOther> other,
+		Func<TOther, TEntity, bool> predicate,
+		Func<TEntity, TKey> keySelector)
+	{
+		return AttachedProperty.GetOrCreate(
+			listState,
+			keySelector,
+			(other, predicate, messenger),
+			CreateRecipient);
+
+		static Recipient<IListState<TEntity>, TEntity, TKey> CreateRecipient(
+			IListState<TEntity> state,
+			Func<TEntity, TKey> keySelector,
+			(IFeed<TOther> other, Func<TOther, TEntity, bool> predicate, IMessenger messenger) config)
+			=> new(state, config.messenger, keySelector, If<IListState<TEntity>, TOther, TEntity, TKey>(config.other, config.predicate, StateExtensions.Update));
+	}
+
+	private static AsyncAction<TState, EntityMessage<TEntity>, Func<TEntity, TKey>> If<TState, TOther, TEntity, TKey>(
+		IFeed<TOther> other,
+		Func<TOther, TEntity, bool> predicate,
+		AsyncAction<TState, EntityMessage<TEntity>, Func<TEntity, TKey>> update)
+		=> async (state, message, keySelector, ct) =>
+		{
+			using var _ = SourceContext.GetOrCreate(other).AsCurrent();
+
+			if ((await other.Option(ct)).IsSome(out var master) && predicate(master, message.Value))
+			{
+				await update(state, message, keySelector, ct);
+			}
+		};
+
+	private sealed class Recipient<TState, TEntity, TKey> : IRecipient<EntityMessage<TEntity>>, IDisposable
```

Comment:
Move to another file?

Reply:
It's private nested class of a pretty small class, I don't see any interest for that